implementation.tex

```diff
@@ -53,15 +51,15 @@ \section{Implementation}\label{sec:implementation}
 Unfortunately several obstacles prevent such an integration from being
 done naturally:
 \begin{itemize}
-  \item The core data structure in incremental plans is the
+  \item The core data structure in incremental circuits is the
```

Comment:
Hmm, you're starting this deep discussion as the last paragraph of the implementation section. I think it's worth breaking it down a bit more:

- Start with motivation: modern databases do not support IVM efficiently. The DBSP algorithm can in principle be used to add such support.
- Both the internal data structures used by modern databases and their external APIs would have to be adjusted.
- Internally, database query engines don't support negative changes
- Externally, IVM requires not just computing the view incrementally, but also reporting the changes to the user as deltas (otherwise even a small updated to the view would require sending its entire snapshot to the user defeating the purpose of incrementality). Today's databases do not expose such an interface, hence the CDC industry....